crates/bevy_core_pipeline/src/bloom/downsampling_pipeline.rs

```diff
@@ -82,28 +82,60 @@ impl FromWorld for BloomDownsamplingPipeline {
             ..Default::default()
         });
 
+        let fullscreen_shader = world.resource::<FullscreenShader>().clone();
+        let fragment_shader = load_embedded_asset!(world, "bloom.wgsl");
+        let base_descriptor = RenderPipelineDescriptor {
+            layout: vec![bind_group_layout.clone()],
+            vertex: fullscreen_shader.to_vertex_state(),
+            fragment: Some(FragmentState {
+                shader: fragment_shader.clone(),
+                targets: vec![Some(ColorTargetState {
+                    format: BLOOM_TEXTURE_FORMAT,
+                    blend: None,
+                    write_mask: ColorWrites::ALL,
+                })],
+                ..default()
+            }),
+            ..default()
+        };
+
+        let specialized_cache =
+            SpecializedCache::new(BloomDownsamplingSpecializer, None, base_descriptor);
+
         BloomDownsamplingPipeline {
             bind_group_layout,
             sampler,
-            fullscreen_shader: world.resource::<FullscreenShader>().clone(),
-            fragment_shader: load_embedded_asset!(world, "bloom.wgsl"),
+            specialized_cache,
         }
     }
 }
 
-impl SpecializedRenderPipeline for BloomDownsamplingPipeline {
-    type Key = BloomDownsamplingPipelineKeys;
+impl Specializer<RenderPipeline> for BloomDownsamplingSpecializer {
+    type Key = BloomDownsamplingKey;
+
+    fn specialize(
+        &self,
+        key: Self::Key,
+        descriptor: &mut RenderPipelineDescriptor,
+    ) -> Result<Canonical<Self::Key>, BevyError> {
+        descriptor.label = Some(if key.first_downsample {
+            "bloom_downsampling_pipeline_first".into()
+        } else {
+            "bloom_downsampling_pipeline".into()
+        });
 
-    fn specialize(&self, key: Self::Key) -> RenderPipelineDescriptor {
-        let layout = vec![self.bind_group_layout.clone()];
+        // TODO: should this error?
```

Comment:
I believe this should be an error yes. Not having a fragment state at this point is not expected and indicates something is very wrong.